search/routes/ui.py

```diff
@@ -130,9 +130,10 @@ def url_for_author_search_builder() -> Dict[str, Callable]:
     """Inject a function to build author name query URLs."""
     def url_for_author_search(forename: str, surname: str) -> str:
         parts = url_parse(url_for('ui.search'))
+        forename_part = ' '.join([part[0] for part in forename.split()])
```

Comment:
If forename is a string, this is building just the first letter of each name separated by spaces. This might not be the pattern to match, especially if someone publishes under `first MI`: "Wells, Herman B".

This seems like it should have a comment too just stating that you're building the abbreviations. Not obvious until reasoning through code.

Reply:
See https://github.com/cul-it/arxiv-search/blob/bug/ARXIVNG-544/search/services/index/authors.py#L53 for how this gets used. The short story is that both "Wells, Herman B." and "Wells, H. B." would respond to a search for "Wells, H. B."